### ISSUE TITLE
Unexpected error codes

### ISSUE THREAD

**Issue by badeend** [WebAssembly/wasi-filesystem#155]
The `error-code` enum contains a few cases that seem odd to me.

First and foremost: 
```
/// Bad descriptor, similar to `EBADF` in POSIX.
bad-descriptor,
```

Assuming the WASI host is implemented correctly, the WASI guest should never be able to observe this, right?

---

Additionally, AFAIK, the following error codes are specific to sockets and do not apply to filesystem operations:
```
/// Connection already in progress, similar to `EALREADY` in POSIX.
already,
/// Operation in progress, similar to `EINPROGRESS` in POSIX.
in-progress,
/// Message too large, similar to `EMSGSIZE` in POSIX.
message-size,
```

**Comment by pchickey** [comment]
> Assuming the WASI host is implemented correctly, the WASI guest should never be able to observe this, right?

Correct - in a correct Component Model implementation, this is not possible to reach. This is a relic of Preview 1 and the prototype period of Preview 2 before resource types were implemented.

> the following error codes are specific to sockets and do not apply to filesystem operations:

Correct, those never got pruned out either.

We should fix these in 0.3. I don't think we have yet figured out our source control strategy is for staging breaking changes that cant land in `main` during the 0.2 series. For lack of a better alternative, you could make a Draft (so it doesnt merge) PR that is called `staging-0.3` or something that has these changes, and we can send more PRs to that branch as we come up with more breaking changes, and periodically merge `main` into it as well so they don't diverge too badly.

**Comment by badeend** [comment]
👍 
Got a PR up at: https://github.com/WebAssembly/wasi-filesystem/pull/156
